examples/platforms/cc2652/crypto/aes_alt.c

```diff
@@ -0,0 +1,194 @@
+        PRCMLoadSet();
+
+        while (!PRCMLoadGet());
+
+    }
+
+    ctx->magic = CC2652_AES_CTX_MAGIC;
+    ctx->key_idx = CC2652_AES_KEY_UNUSED;
+}
+
+void mbedtls_aes_free(mbedtls_aes_context *ctx)
+{
+    otEXPECT(ctx->magic == CC2652_AES_CTX_MAGIC);
+
+    if (ctx->key_idx != CC2652_AES_KEY_UNUSED)
+    {
+        sUsedKeys &= ~(1 << ctx->key_idx);
+    }
+
+    if (--sRefNum == 0)
+    {
+        /* disable the crypto core */
+        /* The TRNG core needs the peripheral power domain powered on to
+         * function. if there is a situation where the power domain must be
+         * powered off, use this code to do so.
+
+        PRCMPowerDomainOff(PRCM_DOMAIN_PERIPH);
+        while (PRCMPowerDomainStatus(PRCM_DOMAIN_PERIPH) != PRCM_DOMAIN_POWER_OFF);
+        */
+        PRCMPeripheralRunDisable(PRCM_PERIPH_CRYPTO);
+        PRCMPeripheralSleepDisable(PRCM_PERIPH_CRYPTO);
+        PRCMPeripheralDeepSleepDisable(PRCM_PERIPH_CRYPTO);
+        PRCMLoadSet();
+
+        while (!PRCMLoadGet());
+
+    }
+
+    memset((void *)ctx, 0x00, sizeof(ctx));
+
+exit:
+    return;
+}
+
+int mbedtls_aes_setkey_enc(mbedtls_aes_context *ctx, const unsigned char *key, unsigned int keybits)
+{
+    unsigned char key_idx;
+    int retval = 0;
+
+    otEXPECT_ACTION(ctx->magic == CC2652_AES_CTX_MAGIC, retval = -1);
+
+    if (ctx->key_idx != CC2652_AES_KEY_UNUSED)
+    {
+        sUsedKeys &= ~(1 << ctx->key_idx);
+    }
+
+    /* our hardware only supports 128 bit keys */
+    otEXPECT_ACTION(keybits == 128u, retval = MBEDTLS_ERR_AES_INVALID_KEY_LENGTH);
+
+    for (key_idx = 0; ((sUsedKeys >> key_idx) & 0x01) != 0 && key_idx < 8; key_idx++);
+
+    /* we have no more room for this key */
+    otEXPECT_ACTION(key_idx < 8, retval = -2);
+
+    otEXPECT_ACTION(CRYPTOAesLoadKey((uint32_t *)key, key_idx) == AES_SUCCESS,
+                    retval = MBEDTLS_ERR_AES_INVALID_KEY_LENGTH);
+
+    sUsedKeys |= (1 << key_idx);
+    ctx->key_idx = key_idx;
+exit:
+    return retval;
+}
+
+int mbedtls_aes_setkey_dec(mbedtls_aes_context *ctx, const unsigned char *key, unsigned int keybits)
+{
+    unsigned char key_idx;
+    int retval = 0;
+
+    otEXPECT_ACTION(ctx->magic == CC2652_AES_CTX_MAGIC, retval = -1);
+
+    if (ctx->key_idx != CC2652_AES_KEY_UNUSED)
+    {
+        sUsedKeys &= ~(1 << ctx->key_idx);
+    }
+
+    /* our hardware only supports 128 bit keys */
+    otEXPECT_ACTION(keybits == 128u, retval = MBEDTLS_ERR_AES_INVALID_KEY_LENGTH);
+
+    for (key_idx = 0; ((sUsedKeys >> key_idx) & 0x01) != 0 && key_idx < 8; key_idx++);
+
+    /* we have no more room for this key */
+    otEXPECT_ACTION(key_idx < 8, retval = -2);
+
+    otEXPECT_ACTION(CRYPTOAesLoadKey((uint32_t *)key, key_idx) == AES_SUCCESS,
+                    retval = MBEDTLS_ERR_AES_INVALID_KEY_LENGTH);
+
+    sUsedKeys |= (1 << key_idx);
+    ctx->key_idx = key_idx;
+exit:
+    return retval;
+}
+
+/**
+ * \brief          AES-ECB block encryption/decryption
```

Comment:
There is a mix of `@` vs. `\` for Doxygen keywords in this PR.  OpenThread typically uses `@`.  Can we unify this PR around `@`?